### PR TITLE
[WebDriverBiDi] refactor test to use `bidi_session.<module>.<command>` syntax

### PR DESCRIPTION
**Pull request by Lightning00Blade**
For the [Serialization options](https://w3c.github.io/webdriver-bidi/#type-script-SerializationOptions) change. The spec defines `null` (in Python `None`) to be accepted value. Use `Undefined` to single don't use.

The other changes uses the `bidi_session` and it's helper to send commands. This is now widely used and provides some type safety when sending command. 

Use `ContextTarget` to abstract the context target creation.

### CODE REVIEW

**Review by sadym-chromium — APPROVED** [review]

**Review by sadym-chromium — APPROVED** [review]

File: tools/webdriver/webdriver/bidi/modules/script.py [hunk 1]
```diff
@@ -78,11 +78,11 @@ def __init__(self, context: str, sandbox: Optional[str] = None):
 class SerializationOptions(Dict[str, Any]):
     def __init__(
             self,
-            max_dom_depth: Optional[int] = None,
+            max_dom_depth: Optional[Union[int, None]] = 0,
```
Comment (whimboo) [inline tools/webdriver/webdriver/bidi/modules/script.py:81]: Shouldn't this be `Union[int, Undefined]`? That would match with the `browsingContext.setViewport` use case. (And yes we should finalize https://github.com/w3c/webdriver-bidi/issues/642).

File: tools/webdriver/webdriver/bidi/modules/script.py [hunk 2]
```diff
@@ -78,11 +78,11 @@ def __init__(self, context: str, sandbox: Optional[str] = None):
...
             max_object_depth: Optional[int] = None,
```
Comment (whimboo) [inline tools/webdriver/webdriver/bidi/modules/script.py:82]: This line and the next need an update as well. Or it's not clear to me based on the description of the PR why you only modified the above one.

File: tools/webdriver/webdriver/bidi/modules/script.py [hunk 3]
```diff
@@ -78,11 +78,11 @@ def __init__(self, context: str, sandbox: Optional[str] = None):
...
             include_shadow_tree: Optional[str] = None
     ):
-        if max_dom_depth is not None:
+        if max_dom_depth != 0:
```
Comment (whimboo) [inline tools/webdriver/webdriver/bidi/modules/script.py:85]: We should not pass this if condition if the value is `None`. Then the optional flag comes into play, and the field should not be added.

File: tools/webdriver/webdriver/bidi/modules/script.py (hunk 1 above)
Comment (Lightning00Blade) [reply tools/webdriver/webdriver/bidi/modules/script.py:81]: I didn't know there was an exception to the status quo. Will update it to match that.

File: tools/webdriver/webdriver/bidi/modules/script.py [hunk 4]
```diff
@@ -78,15 +79,15 @@ def __init__(self, context: str, sandbox: Optional[str] = None):
...
+            include_shadow_tree: Union[Optional[str], Undefined] = UNDEFINED
     ):
-        if max_dom_depth is not None:
+        if max_dom_depth is not UNDEFINED:
             self["maxDomDepth"] = max_dom_depth
-        if max_object_depth is not None:
+        if max_object_depth is not UNDEFINED:
             self["maxObjectDepth"] = max_object_depth
-        if include_shadow_tree is not None:
+        if include_shadow_tree is not UNDEFINED and include_shadow_tree is not None:
```
Comment (Lightning00Blade) [inline tools/webdriver/webdriver/bidi/modules/script.py:90]: Spec does not allow `null` here

File: tools/webdriver/webdriver/bidi/modules/script.py [hunk 5]
```diff
@@ -78,15 +79,15 @@ def __init__(self, context: str, sandbox: Optional[str] = None):
 class SerializationOptions(Dict[str, Any]):
     def __init__(
             self,
-            max_dom_depth: Optional[int] = None,
-            max_object_depth: Optional[int] = None,
-            include_shadow_tree: Optional[str] = None
+            max_dom_depth: Union[Optional[int], Undefined] = UNDEFINED,
```
Comment (whimboo) [inline tools/webdriver/webdriver/bidi/modules/script.py:82]: Something looks wrong here. Similar to the other arguments it should still be optional for both `int` or `Undefined`. Or am I wrong. Maybe @sadym-chromium can have a look at it as well.

```suggestion
            max_dom_depth: Optional[Union[int, Undefined]] = UNDEFINED,
```

Comment (Lightning00Blade) [reply tools/webdriver/webdriver/bidi/modules/script.py:82]: https://github.com/web-platform-tests/wpt/blob/master/tools/webdriver/webdriver/bidi/modules/browsing_context.py#L234

Comment (sadym-chromium) [reply tools/webdriver/webdriver/bidi/modules/script.py:82]: For my understanding the types `Union[Optional[int], Undefined]` and `Optional[Union[int, Undefined]]` are equal, as `Optional` is a union of `type` and `None`. So both would be resolved to `Union[int, Undefined, None]`. I can be wrong here, as not that familiar with Python types.

**Review by sadym-chromium — APPROVED** [review]

File: tools/webdriver/webdriver/bidi/modules/script.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (Lightning00Blade) [reply tools/webdriver/webdriver/bidi/modules/script.py:82]: I would do it in another PR.

**Review by whimboo — APPROVED** [review]
Thanks @sadym-chromium! Looks fine to me as well then. The failures are known and unrelated to these changes. Due to the PR touches the BiDi fixtures all the tests have been run and also caused the Firefox stability job to fail.

Maybe you can explain / update the `comment syntax` in the PR title before merging? I'm not fully sure what that exactly means. Thanks.

**Pull request merged**